Bug title

### Describe the bug

failed to build projects with VS 2019

### Affected OS

- [X] Windows
- [ ] Linux
- [ ] macOS
- [ ] Other (specify below)

### Additional OS information

Visual Studio 2019

### MsQuic version

main

### Steps taken to reproduce bug

Try to build:
cmake --build .
or
build solution from VS

### Expected behavior

successful build

### Actual outcome

A lot of error while trying to compile GNU C extension ```__attribute__((no_instrument_function))```

msquic\src\inc\quic_trace.h(183,29): error C2143: syntax error: missing ')' before '(' (compiling source file \msquic\src\core\ack_tracker.c) [\msquic_cmake\src\core\core.vcxproj]

### Additional details

I need apply this patch to build project with VS2019  (commented ```__attribute__```)

```
-- a/src/inc/quic_trace.h
+++ b/src/inc/quic_trace.h
@@ -175,12 +175,12 @@ extern
 #ifdef __cplusplus
     "C"
 #endif
-char * __attribute__((no_instrument_function))
+char * //__attribute__((no_instrument_function))
 casted_clog_bytearray(const uint8_t * const data,
                       const size_t len,
                       struct clog_param ** head);
 #else
-inline char * __attribute__((no_instrument_function))
+inline char *  // __attribute__((no_instrument_function))
 casted_clog_bytearray(const uint8_t * const data,
                       const size_t len,
                       struct clog_param ** head)
--
```

--- a/src/inc/quic_trace.h
+++ b/src/inc/quic_trace.h
@@ -175,12 +175,18 @@ extern
 #ifdef __cplusplus
     "C"
 #endif
-char * __attribute__((no_instrument_function))
+char *
+#ifndef _WIN32
+__attribute__((no_instrument_function))
+#endif
 casted_clog_bytearray(const uint8_t * const data,
                       const size_t len,
                       struct clog_param ** head);
 #else
-inline char * __attribute__((no_instrument_function))
+inline char *
+#ifndef _WIN32
+__attribute__((no_instrument_function))
+#endif
 casted_clog_bytearray(const uint8_t * const data,
                       const size_t len,
                       struct clog_param ** head)

